nsds_lab_to_nwb/components/tdt/tdt_reader.py

```diff
@@ -0,0 +1,78 @@
+import tdt
+
+class TDTReader():
+    """TDT interface
+    """
+    def __init__(self, path, verbose=False, channels=None):
+        """
+
+        Args:
+            path (str): path to tdt folder
+            verbose (bool, optional): whether to print debugging statements. Defaults to False.
+            channels (list, optional): list of channel ids to import. Defaults to None.
+        """
+        self.path = path
+        self.channels = channels
+        self.verbose = verbose
+        
+        if channels is None:
+            self.tdt_obj = tdt.read_block(path)
+        else:
+            self.tdt_obj = tdt.read_block(path, channel=channels)
+
+        self.streams = self.get_streams()
+        self.block_name = self.tdt_obj['info']['blockname']
+        self.start_time = self.tdt_obj['info']['utc_start_time']
+        
+        if verbose:
+            print('Stream list:')
+            print(self.streams)
+    
+    def get_streams(self):
+        """Get TDT all stream names
+
+        Returns:
+            streams (list): stream names
+        """
+        x = list(self.tdt_obj['streams'].__dict__.items())
+        streams = [e[0] for e in x]
+        return streams
+    
+    def get_metadata(self, stream):
+        """Get specified stream metadata
+
+        Args:
+            stream (string): stream name
+
+        Returns:
+            meta (dict): dictionary containing stream recording parameters
+        """
+        meta = {}
+        meta['sample_rate'] = self.tdt_obj['streams'][stream]['fs']
+        meta['channel_ids'] = self.tdt_obj['streams'][stream]['channel']
+        meta['num_channels'], meta['num_samples'] = self.tdt_obj['streams'][stream]['data'].shape
```

Comment:
I think streams like `aud0` and `mrk1` are 1d so don't have a channels. Probably need to case this.

Reply:
I added logic to return num_channels = 1 if the stream data is a 1d array